File: python/027_perfect-numbers/perfect_numbers.py
def classify(number):
    if number > 0:        
        salida = ['perfect','abundant','deficient']
        suma = 0
        for i in range(1,number//2+1):
            suma += 0 if number % i else i
        suma -= number    
        return salida[suma//abs(suma) if suma else 0]
    raise ValueError("Classification is only possible for positive integers.")

File: javascript/060_perfect-numbers/perfect-numbers.js
export const classify = (number) => {
  if(number <= 0) throw new Error('Classification is only possible for natural numbers.');
  let suma = 0;
  for (let i = 1; i <= number / 2; i++)
    suma += number % i ? 0 : i;
  return suma == number ? 'perfect' : suma > number ? 'abundant' : 'deficient';
};
